java/statestore/src/test/java/sleeper/statestore/transactionlog/DeleteTransactionLogSnapshotsIT.java

```diff
@@ -0,0 +1,198 @@
+import static sleeper.configuration.properties.instance.CommonProperty.FILE_SYSTEM;
+import static sleeper.configuration.properties.table.TablePropertiesTestHelper.createTestTableProperties;
+import static sleeper.configuration.properties.table.TableProperty.STATESTORE_CLASSNAME;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_ID;
+import static sleeper.configuration.properties.table.TableProperty.TABLE_NAME;
+import static sleeper.configuration.properties.table.TableProperty.TRANSACTION_LOG_SNAPSHOT_EXPIRY_IN_DAYS;
+import static sleeper.core.schema.SchemaTestHelper.schemaWithKey;
+
+public class DeleteTransactionLogSnapshotsIT extends TransactionLogStateStoreTestBase {
+    @TempDir
+    private java.nio.file.Path tempDir;
+    private final Schema schema = schemaWithKey("key", new LongType());
+    private FileSystem fs;
+
+    @BeforeEach
+    public void setup() throws IOException {
+        instanceProperties.set(FILE_SYSTEM, "file://");
+        instanceProperties.set(DATA_BUCKET, tempDir.toString());
+        fs = FileSystem.get(configuration);
+    }
+
+    @Test
+    void shouldDeleteOldFilesSnapshotsForOneTable() throws Exception {
+        // Given
+        TableProperties table = createTable("test-table-id-1", "test-table-1");
+        table.setNumber(TRANSACTION_LOG_SNAPSHOT_EXPIRY_IN_DAYS, 1);
+        StateStore stateStore = createStateStore(table);
+        PartitionsBuilder partitionsBuilder = new PartitionsBuilder(schema).rootFirst("root");
+        stateStore.initialise(partitionsBuilder.buildList());
+        FileReferenceFactory factory = FileReferenceFactory.from(stateStore);
+        stateStore.addFile(factory.rootFile("test1.parquet", 123L));
+        createSnapshotsAt(table, Instant.parse("2024-04-25T11:24:00Z"));
+        stateStore.clearFileData();
+        stateStore.initialise(partitionsBuilder.splitToNewChildren("root", "L", "R", 123L).buildList());
+        factory = FileReferenceFactory.from(stateStore);
+        stateStore.addFile(factory.rootFile("test3.parquet", 789L));
+        createSnapshotsAt(table, Instant.parse("2024-04-27T11:24:00Z"));
+
+        // When
+        deleteSnapshotsAt(table, Instant.parse("2024-04-27T11:24:00Z"));
+
+        // Then
+        assertThat(snapshotStore(table).getLatestSnapshots())
+                .isEqualTo(new LatestSnapshots(
+                        filesSnapshot(table, 3),
+                        partitionsSnapshot(table, 2)));
+        assertThat(snapshotStore(table).getFilesSnapshots())
+                .containsExactly(filesSnapshot(table, 3));
+        assertThat(snapshotStore(table).getPartitionsSnapshots())
+                .containsExactly(partitionsSnapshot(table, 2));
+        assertThat(fs.exists(new Path(filesSnapshot(table, 1).getPath()))).isFalse();
+        assertThat(fs.exists(new Path(partitionsSnapshot(table, 1).getPath()))).isFalse();
```

Comment:
This isn't asserting that the files weren't deleted for the snapshot that wasn't old enough to be deleted. Maybe this should be two or three tests, one for an old snapshot, one for one that isn't old enough, and one for multiple snapshots? It's a bit difficult to see that this is testing all three of those.